src/ucp/wireup/wireup_cm.c

```diff
@@ -441,19 +447,66 @@ static unsigned ucp_ep_cm_remote_disconnect_progress(ucp_ep_h ucp_ep)
     return 1;
 }
 
+/*
+ * The main thread progress part of connection establishment on server side
+ */
+static unsigned ucp_cm_server_connect_progress(void *arg)
+{
+    ucp_ep_h ucp_ep = arg;
+
+    UCS_ASYNC_BLOCK(&ucp_ep->worker->async);
+    ucs_debug("ep %p: CM server connect progress, flags 0x%x", ucp_ep,
+              ucp_ep->flags);
+    ucp_wireup_remote_connected(ucp_ep);
+    UCS_ASYNC_UNBLOCK(&ucp_ep->worker->async);
+    return 1;
+}
+
+static int ucp_ep_conn_cb_cmp(const ucs_callbackq_elem_t *elem, void *arg)
+{
+    ucp_ep_h ep = arg;
+    ucp_cm_client_connect_progress_arg_t *client_progress_arg;
+
+    if (elem->cb == ucp_cm_client_connect_progress) {
+        client_progress_arg = elem->arg;
```

Comment:
client_progress_arg var is redundant

Reply:
Do you mean `return ((ucp_cm_client_connect_progress_arg_t *)elem->arg)->ucp_ep == ep;` is better? I don't think so...